spanner/integration_test.go

```diff
@@ -2013,6 +2019,119 @@ func TestIntegration_BasicTypes(t *testing.T) {
 	}
 }
 
+// Test encoding/decoding non-struct Cloud Spanner Proto Column types.
+func TestIntegration_BasicTypes_ProtoColumns(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+	stmts := []string{}
+	client, _, cleanup := prepareIntegrationTestForProtoColumns(ctx, t, DefaultSessionPoolConfig, stmts)
+	defer cleanup()
+
+	singerProtoEnum := pb.Genre_ROCK
+	singerProtoMessage := pb.SingerInfo{
+		SingerId:    proto.Int64(1),
+		BirthDate:   proto.String("January"),
+		Nationality: proto.String("Country1"),
+		Genre:       &singerProtoEnum,
+	}
+	bytesSingerProtoMessage, _ := proto.Marshal(&singerProtoMessage)
+
+	tests := []struct {
+		col  string
+		val  interface{}
+		want interface{}
+	}{
+		{col: "ProtoMessage", val: &singerProtoMessage, want: singerProtoMessage},
+		{col: "ProtoEnum", val: pb.Genre_ROCK, want: singerProtoEnum},
```

Comment:
This is to differentiate the data passed in `val` and `want`. `pb.Genre_ROCK` is passed in `val`, and `singerProtoEnum` is passed in `want` in the tests. Also in some places we need to pass a pointer instead of just the constant. That is why we need both `pb.Genre_ROCK` and `singerProtoEnum`